compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -402,11 +409,10 @@ static LogicalResult setDefaultRootConfig(
                          : DispatchLoweringPassPipeline::CPUBufferOpsDefault);
 }
 
-static LogicalResult setSandboxRootConfig(func::FuncOp entryPointFn,
-                                          linalg::ContractionOpInterface op,
-                                          ArrayRef<int64_t> flowTileSizes,
-                                          ArrayRef<int64_t> target2ndTileSizes,
-                                          int vectorSize) {
+static LogicalResult setMatmulPadRootConfig(
```

Comment:
Could we keep the reference/fallback version in a separate function and then just use the flag to invoke the reference version of the padding version? I have to introduce another version for peeling and with this approach things won't compose well. That would also make this version less branchy. We would duplicate some code but I think it would look a bit cleaner overall.

Reply:
SGTM. :)